test/unit/lib/plugins/aws/provider.test.js

```diff
@@ -335,6 +329,66 @@ describe('AwsProvider', () => {
 });
 
 describe('test/unit/lib/plugins/aws/provider.test.js', () => {
+  describe('#getProviderName and #sessionCache', () => {
+    let sls;
+    const expectedToken = '123';
+
+    before(async () => {
+      // Fake service that update credentials
+      class FakeCloudFormation {
+        constructor(credentials) {
+          this.credentials = credentials;
+          this.credentials.credentials.sessionToken = expectedToken;
+        }
+        describeStacks() {
+          return { promise: async () => {} };
+        }
+      }
+      // Stub functions for the credentials creation in the provider
+      class SharedIniFileCredentials {
+        constructor() {
+          this.sessionToken = 'abc';
+          this.accessKeyId = 'keyId';
+          this.secretAccessKey = 'secret';
+        }
+      }
+      class EnvironmentCredentials {
+        constructor() {
+          this.sessionToken = 'env';
+          this.accessKeyId = 'keyId';
+          this.secretAccessKey = 'secret';
+        }
+      }
+      const modulesCacheStub = {
+        'aws-sdk': {
+          SharedIniFileCredentials,
+          EnvironmentCredentials,
+          CloudFormation: FakeCloudFormation,
+        },
+      };
+      const { serverless } = await runServerless({
+        fixture: 'aws',
+        command: 'print',
+        modulesCacheStub,
+      });
+      sls = serverless;
+    });
+
+    it('`AwsProvider.getProviderName()` should resolve provider name', () => {
+      expect(AwsProvider.getProviderName()).to.equal('aws');
```

Comment:
I believe it should use `sls.getProvider('aws').getProviderName()` instead of current notation

Reply:
Unfortunately as `getProviderName()` is static, it can only be called on the class (if I understood properly: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static). I've tried your notation and only could obtain an `undefined is not a function` error :cry: . If you have any pointer on how to resolve that, I would be more than happy to use them. I find it a bit disturbing that we don't use the sls too.

Reply:
I totally missed that - in such case I guess we can stick to current approach to avoid additional refactorings.